visualization/app/codeCharta/util/gameObjectsParser/gameObjectsImporter.ts

```diff
@@ -0,0 +1,137 @@
+import md5 from "md5"
+import { AttributeTypes, CodeMapNode, Edge, FixedPosition, NodeType } from "../../codeCharta.model"
+import { ExportWrappedCCFile } from "../../codeCharta.api.model"
+
+export function parseGameObjectsFile(data) {
+	const { gameObjectPositions, cycles = [] } = JSON.parse(data)
+
+	const ccJson: ExportWrappedCCFile = {
+		checksum: "",
+		data: {
+			projectName: "ScriptProjectName",
+			fileChecksum: "",
+			apiVersion: "1.3",
+			nodes: []
+		}
+	}
+
+	const nodes = [{ name: "base", type: NodeType.FOLDER, attributes: {}, children: [] }]
+	gameObjectPositions.push(createBaseGameObjectPosition(gameObjectPositions))
+
+	for (const gameObjectPosition of gameObjectPositions) {
+		const nodeNames = gameObjectPosition.name.split(".")
+		if (nodeNames[0] !== "base") {
+			addNodeRecursively(nodeNames, nodes[0].children, "base", gameObjectPosition, gameObjectPositions)
+		}
+	}
+
+	ccJson.data.nodes = nodes
+	ccJson.data.edges = cycles.map(cycle => createEdge(cycle))
+	ccJson.data.attributeTypes = createAttributeTypes()
+	ccJson.checksum = md5(JSON.stringify(ccJson.data))
+	return ccJson
+}
+
+function addNodeRecursively(nodeNames: string[], nodes: CodeMapNode[], parentNodeName: string, gameObjectPosition, gameObjectPositions) {
+	if (nodeNames.length === 0) return
+
+	// get current node name and create cc node structure
+	const [nodeName] = nodeNames
+	let node: CodeMapNode = {
+		name: nodeName,
+		type: isFile(nodeNames) ? NodeType.FILE : NodeType.FOLDER,
+		attributes: {}
+	}
+	if (!isFile(nodeNames)) node.children = []
+	if (isFile(nodeNames)) node = wrapFileInAFolder(nodeName, node, gameObjectPosition)
```

Comment:
```suggestion
	if (!isFile(nodeNames)) node.children = []
	else node = wrapFileInAFolder(nodeName, node, gameObjectPosition)
```

Reply:
That's not how I see it. In principle you're right but in this case, we don't return anything, so we can't use the advantage of an early return. Additionally, it's not necessary to check if `nodeNames` is a file when we checked the other case before.